pkg/controller/controller.go

```diff
@@ -348,3 +360,34 @@ func (r *remoteCluster) serviceDeleted(key string) error {
 	}
 	return nil
 }
+
+func (r *remoteCluster) canFederateService(service string) bool {
+	return !skipServices[service]
+}
+
+func (r *remoteCluster) canFederateNamespace(namespace string) bool {
+	if !skipNamespaces[namespace] {
+		for _, element := range namespaceExclusionList {
+			if strings.Contains(namespace, element) {
+				return false
+			}
+		}
+		return true
+	}
+	return false
+}
+
+func (r *remoteCluster) federateNamespace(namespace string) {
+	ns, err := r.clientset.CoreV1().Namespaces().Get(namespace, metav1.GetOptions{})
+	if err != nil {
+		klog.Warningf("Failed to get namespace %s details from cluster  %s", namespace, r.clusterID)
+		return
+	}
+	ns.SetNamespace(namespace)
+	err = r.controller.federator.Distribute(ns)
+	if err != nil {
+		klog.Warningf("Failed to federate namespace %s", namespace)
+	}
+	skipNamespaces[namespace] = true
```

Comment:
It may be possible for `skipNamespaces` to be accessed by multiple threads so probably should be protected to be on the safe side, either via a lock or sync.Map.

That said, do we really need `skipNamespaces`, ie is it OK to re-federate a namespace? I mean that will happen on restart of the controller anyway.

Reply:
It is more of an optimization. We're already maintaining a list of skipped namespaces, might as well use it to track ones already federated.